### PR TITLE
Port window-frame

### PR DESCRIPTION
**Pull request by shaleh**
Add a `frame` method to `LispWindowRef` to get the window's frame.

### CODE REVIEW

**Comment by shaleh** [comment]
The `frame` method will reduce redundancy as port the rest of the frame based functions in window.c.

File: rust_src/src/windows.rs [hunk 1]
```diff
@@ -179,3 +184,17 @@ pub fn minibuffer_selected_window() -> LispObject {
...
+#[lisp_fn(min = "0")]
+pub fn window_frame(window: LispObject) -> LispObject {
+    let win = if window.is_nil() {
+        selected_window()
+    } else {
+        window
+    };
+
+    win.as_window_or_error().frame()
+}
```
Comment (brotzeit) [inline rust_src/src/windows.rs]: It looks like the c code makes use of `CHECK_VALID_WINDOW` which throws a `Qwindow_valid_p` error. We don't have this implemented yet, but I wonder if we already have ported functions with `decode_valid_window` incorrectly, hmmm.

Comment (shaleh) [reply rust_src/src/windows.rs]: I have a PR almost ready :-)

Comment (shaleh) [reply rust_src/src/windows.rs]: The window-buffer routines also should have used `CHECK_VALID`

Comment (shaleh) [reply rust_src/src/windows.rs]: I am adding `window_valid_or_selected` and `window_or_selected` to windows.rs.

Comment (shaleh) [reply rust_src/src/windows.rs]: Bike shed while I am away. I will commit this afternoon.

**Comment by shaleh** [comment]
Waiting on PR #411

**Pull request merged**